core/common/test/samples/InstantSamples.kt

```diff
@@ -0,0 +1,360 @@
+        // Converting an Instant to the number of milliseconds since the Unix epoch
+        check(Instant.fromEpochMilliseconds(0).toEpochMilliseconds() == 0L)
+        check(Instant.fromEpochMilliseconds(1_000_000_000_123).toEpochMilliseconds() == 1_000_000_000_123L)
+        check(Instant.fromEpochSeconds(1_000_000_000, nanosecondAdjustment = 123_999_999)
+            .toEpochMilliseconds() == 1_000_000_000_123L)
+    }
+
+    @Test
+    fun plusDuration() {
+        // Finding a moment that's later than the starting point by the given amount of real time
+        val instant = Instant.fromEpochMilliseconds(epochMilliseconds = 7 * 60 * 60 * 1000)
+        val fiveHoursLater = instant + 5.hours
+        check(fiveHoursLater.toEpochMilliseconds() == 12 * 60 * 60 * 1000L)
+    }
+
+    @Test
+    fun minusDuration() {
+        // Finding a moment that's earlier than the starting point by the given amount of real time
+        val instant = Instant.fromEpochMilliseconds(epochMilliseconds = 7 * 60 * 60 * 1000)
+        val fiveHoursEarlier = instant - 5.hours
+        check(fiveHoursEarlier.toEpochMilliseconds() == 2 * 60 * 60 * 1000L)
+    }
+
+    @Test
+    fun minusInstant() {
+        // Finding the difference between two instants in terms of elapsed time
+        check(Instant.fromEpochSeconds(0) - Instant.fromEpochSeconds(epochSeconds = 7 * 60 * 60) == (-7).hours)
+    }
+
+    @Test
+    fun compareToSample() {
+        // Finding out which of two instants is earlier
+        fun randomInstant() = Instant.fromEpochMilliseconds(
+            Random.nextLong(Instant.DISTANT_PAST.toEpochMilliseconds(), Instant.DISTANT_FUTURE.toEpochMilliseconds())
+        )
+        repeat(100) {
+            val instant1 = randomInstant()
+            val instant2 = randomInstant()
+            check((instant1 < instant2) == (instant1.toEpochMilliseconds() < instant2.toEpochMilliseconds()))
+        }
+    }
+
+    @Test
+    fun toStringSample() {
+        // Converting an Instant to a string
+        check(Instant.fromEpochMilliseconds(0).toString() == "1970-01-01T00:00:00Z")
+    }
+
+    @Test
+    fun fromEpochMilliseconds() {
+        // Constructing an Instant from the number of milliseconds since the Unix epoch
+        check(Instant.fromEpochMilliseconds(epochMilliseconds = 0) == Instant.parse("1970-01-01T00:00:00Z"))
+        check(Instant.fromEpochMilliseconds(epochMilliseconds = 1_000_000_000_123)
+            == Instant.parse("2001-09-09T01:46:40.123Z"))
+    }
+
+    @Test
+    fun fromEpochSeconds() {
+        // Constructing an Instant from the number of seconds and nanoseconds since the Unix epoch
+        check(Instant.fromEpochSeconds(epochSeconds = 0) == Instant.parse("1970-01-01T00:00:00Z"))
+        check(Instant.fromEpochSeconds(epochSeconds = 1_000_001_234, nanosecondAdjustment = -1_234_000_000_001)
+            == Instant.parse("2001-09-09T01:46:39.999999999Z"))
+    }
+
+    @Test
+    fun fromEpochSecondsIntNanos() {
```

Comment:
I believe it can be merged with the sample above and an Int overload is enough to show in the sample.

Reply:
I don't understand the benefit of merging together samples for several overloads. Here are several downsides:
* This page https://kotlinlang.org/api/kotlinx-datetime/kotlinx-datetime/kotlinx.datetime/plus.html would contain much more repeated information than it already has.
* The point the sample is trying to make gets muddied by including irrelevant code (in this case, code for the other overload).
* It's more difficult to evolve the samples separately. If we learn of nice specific examples where `Long` is more common than `Int`, we can easily incorporate them with the current sample-per-overload scheme.

Reply:
Oh, ok, I didn't notice that the overloads are merged. This is an important point.